schematics/component/index.ts

```diff
@@ -27,7 +27,79 @@ function addImportToNgModule(options: ComponentOptions): Rule {
     if (!options.module) {
       return host;
     }
+    if (!options.createModule && options.module) {
+      addImportToDeclarations(host, options);
+    }
+    if (options.createModule && options.module) {
+      addImportToImports(host, options);
+    }
+    return host;
+  };
+}
+
+function addImportToDeclarations(host: Tree, options: ComponentOptions): void {
+  if (options.module) {
+    const modulePath = options.module;
+    let source = readIntoSourceFile(host, modulePath);
+
+    const componentPath = `/${options.path}/`
+                          + (options.flat ? '' : strings.dasherize(options.name) + '/')
+                          + strings.dasherize(options.name)
+                          + '.component';
+    const relativePath = buildRelativePath(modulePath, componentPath);
+    const classifiedName = strings.classify(`${options.name}Component`);
+    const declarationChanges = addDeclarationToModule(source,
+                                                      modulePath,
+                                                      classifiedName,
+                                                      relativePath);
+
+    const declarationRecorder = host.beginUpdate(modulePath);
+    for (const change of declarationChanges) {
+      if (change instanceof InsertChange) {
+        declarationRecorder.insertLeft(change.pos, change.toAdd);
+      }
+    }
+    host.commitUpdate(declarationRecorder);
+
+    if (options.export) {
+      // Need to refresh the AST because we overwrote the file in the host.
+      source = readIntoSourceFile(host, modulePath);
 
+      const exportRecorder = host.beginUpdate(modulePath);
+      const exportChanges = addExportToModule(source, modulePath,
+                                              strings.classify(`${options.name}Component`),
+                                              relativePath);
+
+      for (const change of exportChanges) {
+        if (change instanceof InsertChange) {
+          exportRecorder.insertLeft(change.pos, change.toAdd);
+        }
+      }
+      host.commitUpdate(exportRecorder);
+    }
+
+    if (options.entryComponent) {
+      // Need to refresh the AST because we overwrote the file in the host.
```

Comment:
Can't you just commit the update later? Or does this get a bit messy?

Reply:
This was copied over from the angular schematic, so Im assuming it's needed. Probably to make sure that host is properly updated and not overwriting it.